client/my-sites/plans/jetpack-plans/plan-upgrade/index.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getItemSlugByDuration } from './utils';
+import { INTRO_PRICING_DISCOUNT_PERCENTAGE } from '../constants';
+import ProductCard from '../product-card';
+import ProductGridSection from '../product-grid/section';
+import { slugToSelectorProduct } from '../utils';
+import { getCurrentUserCurrencyCode } from 'calypso/state/current-user/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+/**
+ * Type dependencies
+ */
+import type { PlanRecommendation } from './types';
+import type { Duration, PurchaseCallback, SelectorProduct } from '../types';
+
+type Props = {
+	planRecommendation: PlanRecommendation;
+	duration: Duration;
+	onSelectProduct: PurchaseCallback;
+};
+
+const PlanUpgradeSection: React.FC< Props > = ( {
+	planRecommendation,
+	duration,
+	onSelectProduct,
+} ) => {
+	const translate = useTranslate();
+	const siteId = useSelector( getSelectedSiteId );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+
+	let [ legacyPlan, newPlans ] = planRecommendation;
+
+	legacyPlan = getItemSlugByDuration( legacyPlan, duration );
+	newPlans = newPlans.map( ( p ) => getItemSlugByDuration( p, duration ) );
+
+	const legacyItem = slugToSelectorProduct( legacyPlan ) as SelectorProduct;
+	const newItems = newPlans.map( slugToSelectorProduct ) as SelectorProduct[];
+	const newItemCount = newItems.length;
+
+	return (
+		<ProductGridSection
+			title={ translate( 'We recommend {{product/}} based on your selection', {
+				components: {
+					product: (
+						<span className="plan-upgrade__product">
+							{ newItems.reduce( ( result, { displayName }, index ) => {
```

Comment:
Separate names by commas and a &. `displayName` is not necessarily a string, so using `join` wasn't possible.